master/buildbot/process/build.py

```diff
@@ -190,25 +194,23 @@ def getWorkerName(self):
         return self.workerforbuilder.worker.workername
     deprecatedWorkerClassMethod(locals(), getWorkerName)
 
-    def setupProperties(self):
-        props = interfaces.IProperties(self)
-
-        # give the properties a reference back to this build
-        props.build = self
+    @staticmethod
+    def setupProperties(props, requests, builder):
 
         # start with global properties from the configuration
-        props.updateFromProperties(self.master.config.properties)
+        props.updateFromProperties(builder.master.config.properties)
 
         # from the SourceStamps, which have properties via Change
-        for change in self.allChanges():
+        sources = requests[0].mergeSourceStampsWith(requests[1:])
```

Comment:
This is the way we compute Build sources in its __init__. If there are more places this is used we could move the sources setup to a separate function, but for now I think the amount of duplication of logic is too small.